src/components/Explore/SearchBar.tsx

```diff
@@ -1,60 +1,74 @@
-import useTheme from 'hooks/useTheme'
 import { useAtom } from 'jotai'
 import { useState } from 'react'
-import { Search } from 'react-feather'
 import styled from 'styled-components/macro'
 
+import searchIcon from './search.svg'
 import { filterStringAtom } from './state'
+import xIcon from './x.svg'
 
 export const SMALL_MEDIA_BREAKPOINT = '580px'
+const ICON_SIZE = '20px'
 
-const StyledSearchBar = styled.div<{ focused: boolean }>`
+const SearchBarContainer = styled.div`
   display: flex;
   flex: 1;
-  align-items: center;
-  gap: 8px;
-  border-radius: 12px;
-  background-color: ${({ theme, focused }) => (focused ? theme.bg2 : theme.bg0)};
-  color: ${({ theme }) => theme.text2};
-  font-size: 16px;
-  padding: 0px 12px;
 `
-const SearchInput = styled.input`
-  background-color: transparent;
+
+const SearchInput = styled.input<{ expanded: boolean }>`
+  background: no-repeat scroll 7px 7px;
+  background-image: url(${searchIcon});
+  background-size: 20px 20px;
+  background-position: 12px center;
+  background-color: ${({ theme }) => theme.bg0};
+  left: 10px;
+  border-radius: 12px;
   border: none;
-  width: 90%;
+  height: 100%;
+  width: ${({ expanded }) => (expanded ? '100%' : '44px')};
   font-size: 16px;
+  padding-left: 40px;
   color: ${({ theme }) => theme.text2};
+  transition: width 0.75s cubic-bezier(0, 0.795, 0, 1);
 
   :focus {
     outline: none;
     background-color: ${({ theme }) => theme.bg2};
   }
   ::placeholder {
-    color: ${({ theme }) => theme.text3};
+    color: ${({ theme, expanded }) => (expanded ? theme.text3 : 'transparent')};
     @media only screen and (max-width: ${SMALL_MEDIA_BREAKPOINT}) {
       color: transparent;
     }
   }
+  ::-webkit-search-cancel-button {
+    -webkit-appearance: none;
+    appearance: none;
+    height: ${ICON_SIZE};
+    width: ${ICON_SIZE};
+    background-image: url(${xIcon});
+    margin-right: 10px;
+    background-size: ${ICON_SIZE} ${ICON_SIZE};
+    cursor: pointer;
+  }
 `
 
 export default function SearchBar() {
-  const theme = useTheme()
-  const [isFocused, setFocused] = useState(false)
   const [filterString, setFilterString] = useAtom(filterStringAtom)
+  const [isExpanded, setExpanded] = useState(false)
 
   return (
-    <StyledSearchBar focused={isFocused}>
-      <Search size={20} color={theme.text3} />
+    <SearchBarContainer>
       <SearchInput
-        type="text"
-        value={filterString}
-        onChange={({ target: { value } }) => setFilterString(value)}
+        expanded={isExpanded}
+        type="search"
         placeholder="Search token or paste address"
         id="searchBar"
-        onFocus={() => setFocused(true)}
-        onBlur={() => setFocused(false)}
+        onBlur={() => isExpanded && filterString.length === 0 && setExpanded(false)}
```

Comment:
shouldn't need to manually setExpanded

Reply:
I think I need isExpanded for the check on whether filterstring is size 0 (if filterstring isn't of length 0 then it shouldn't collapse)